### DISCUSSION TITLE
S40 Smart Thermostat Compatibility

### DISCUSSION THREAD

**Discussion by tedski** [PeteRager/lennoxs30#226] [Q&A]
Lennox has announced the successor to the S30 iComfort series to be launched in early-March 2023.  From the limited literature and sales announcements I can find thus far, it seems to be very similar to the S30 with some feature adds.  One of the most attractive additions is support for remote room sensors to do zone averaging.  More can be found [on the Lennox press release](https://www.lennox.com/lennox-life/comfort-matters/lennox-innovation/unlock-ultimate-comfort-with-the-innovative-lennox-s40-smart-thermostat-and-accessories).

I don't expect much to be known at this point about whether the API will be the same, but I wanted to start the discussion to get information as early as possible as techs start getting hands on with these units.  Does anyone already know how these might integrate with the iComfort cloud or local APIs that this library already uses?

**Comment by PeteRager** [comment]
Thanks for posting.  First I’ve heard about it.  Hopefully it works similar to an S30.

**Comment by PeteRager** [comment]
here’s the manual, looks like s30 + a few new things.  

https://www.lennoxcommercial.com/dA/82d822f66e/file/1070920~508223-01.pdf
https://www.lennoxcommercial.com/dA/47f1fa7a9f/file/1070962~508222-01.pdf

**Comment by ghwong** [comment]
I just recently replace my whole HVAC system with EL22XPV + SL280 + S40
I install and setup HA and connect it local and the API seem to work on the basic thermostat reading and parameters. It didn't work for the heat pump and furnace (indoor/outdoor) parameter.
How can I help on pulling message logs to help get this working?

**Comment by PeteRager** [comment]
Thanks for the note, glad the basics are working.

Follow the instructions here to comfigurethe message logging.  Then restart HASS and the log file should be in the same spot as the home-assistant.log file.   Post that here or I can provide email to send it to.  I can load that into a simulator.

Regarding the parameters what are you seeing or not seeing? Did the devices get created?  If you click on the devices is there any diagnostic data?  Also check that you enabled the diagnostic sensors in the configuration.

**Reply by ghwong** [reply]
The devices did get created, I can only see basic zone temperature and setpoints. All the devices diagnostic data are unavailable or not showing.

**Comment by PeteRager** [comment]
Did you set the diagnostic level to 2?

https://github.com/PeteRager/lennoxs30#diagnostic_level

The file sent is not the message log.  Inside of the integration configuration check the box "Log s30 messages to a separate file", then enter a file name.  In this screen shot it's called s30_local.log.  

![4199570C-9D73-4169-9EED-88C30B0ADD67](https://user-images.githubusercontent.com/76050312/231829008-784852e0-6f09-4f17-acdc-c5cac918b694.jpeg)

**Reply by ghwong** [reply]
Got it.,.. here is the correct log file you requested 
[s30_local.log](https://github.com/PeteRager/lennoxs30/files/11224541/s30_local.log)

**Comment by PeteRager** [comment]
Thanks that what we needed.  I do see that the diagControl value is set to zero.  Do you have a number entity

number.my home.diag_level 

When this is zero no diagnostic data is sent.  Set that to two.

I can see you are getting data from the zone and the outdoor temperature sensor.

**Reply by ghwong** [reply]
how can I change that? is that in hass settings?

**Reply by ghwong** [reply]
my bad.... I overlook that settings. I change it to 2 and now all diag value are showing now.
seem like what you have is compatible with S40.
I added the new logs for reference
[s30_local (1).log](https://github.com/PeteRager/lennoxs30/files/11225001/s30_local.1.log)

**Reply by ghwong** [reply]
the only thing that we are not seeing on the new S40 is my remote room sensors

**Reply by PeteRager** [reply]
Sounds good.  That makes sense as the S30 did not have remote sensors and that will need to be added

Looking at the message log file, it's looks like you have 1 zone.  How many remote sensors do you have?  Do they have names?

**Reply by ghwong** [reply]
Yes. I have only 1 zone and 2 remote sensor named "Master Bedroom" and "Basement"
The model number is 22V25 for both of them

**Reply by PeteRager** [reply]
I do see some references for the firmware for the 22v25, but no data for them in the message log.  I also don't see the names in there either.  I suspect that Lennox has added a new topic that we need to subscribe to in order to get the data, however we need to determine what that is.

Go ahead and create an issue for this item and we can track it there.
